example/server_transform/server_transform.c

```diff
@@ -237,6 +237,7 @@ transform_read(TSCont contp, TransformData *data)
   data->input_reader = NULL;
 
   data->server_vio = TSVConnRead(data->server_vc, contp, data->output_buf, data->content_length);
+  TSIOBufferWaterMarkSet( data->output_buf, data->content_length - sizeof( int ) ); // for performance
```

Comment:
I think this is wrong.  I think we are still in the process of reading in the full content from the request client.  So the content_length would still be zero here.  It is set after the read from the transform completes:

https://github.com/apache/trafficserver/blob/b6b61fbd3e232dbef010dc8b712bb2076510bd7f/example/plugins/c-api/server_transform/server_transform.c#L151